AMQP update. (#526)
mulesoft/docs-connectors: modules/ROOT/pages/amqp/amqp-consume.adoc

:imagesdir: ../../assets/images/

> The `consume` operation in AMQP Connector provides the ability to consume a message at any given time of the flow, from any given AMQP queue.

Reviewer: ```suggestion
The `consume` operation in AMQP Connector provides the ability to consume a message at any given time in the flow, from any given AMQP queue.
```

== Consume a Message

The syntax to consume a message from a queue is:

[source,xml,linenums]
----
<amqp:consume config-ref="AMQP_config" queueName="#[vars.queue]"/>
----

> This operation consumes the first available message in the queue identified by the `queueName`, and converts it to an `AmqpMessage`, which results in the following structure:

Reviewer: ```suggestion
This operation consumes the first available message in the queue identified by the `queueName` value, and converts it to an `AmqpMessage` value, which results in the following structure:
```

* The message's content as payload
* The message's metadata in the message attributes

> Once received, the message is immediately ACKed by default. If you want to control the ACK of the message after some processing, then set the `ackMode` to `MANUAL`.

Reviewer: ```suggestion
After it is received, the message is immediately acknowledged by default. If you want to control ACK timing, then set `ackMode` to `MANUAL`.
```

> For more information about message ACKs, see xref:amqp/amqp-ack.adoc[How to handle Acknowledgment].

Reviewer: ```suggestion
For more information about message ACK operations, see xref:amqp/amqp-ack.adoc[How to Handle Acknowledgment].
```

== Wait for a Message

By default, the maximum wait time is configured to 10 seconds. If no message is available during the specified wait time, an `AMQP:TIMEOUT` error is thrown. You can customize the wait time by configuring the `maximumWait` and `maximumWaitUnit` parameters.

> To wait indefinitely for a Message to arrive, the `maximumWait` value must be set to `-1`. In this case, no `TIMEOUT` error will be thrown.

Reviewer: ```suggestion
To create an indefinite wait time for a message to arrive, set the `maximumWait` value to `-1`. In this case, no `TIMEOUT` error is thrown.
```

== MIME Types and Encoding

> AMQP Connector does its best to auto determine a message’s MIME type (`contentType`) based on the `contentType` property of the message. However, there are cases in which that best guess is not enough, and you need firsthand knowledge of the message’s content.

Reviewer: ```suggestion
AMQP Connector is designed to auto-determine a message’s MIME type (`contentType`) based on the `contentType` property of the message. However, there are cases in which it cannot make this determination, and you need firsthand knowledge of the message’s content.
```

In such cases, you can force the content type to a particular value by using the `contentType` parameter.

> The same process works for encoding. By default, the connector assumes that the Mule's default encoding matches the one in the message if no other information is provided. You can set this by using the `encoding` parameter.

Reviewer: ```suggestion
The same process works for encoding. By default, the connector assumes that the default encoding of the Mule instance matches that of the message, if no other information is provided. You can set this by using the `encoding` parameter.
```

== Incoming Message Metadata

> As stated before, each message received consists of two parts:

Reviewer: ```suggestion
As stated earlier, each message received consists of two parts:
```
